File: src/main/webui/src/data/FileResponse.ts

export interface GetFilesResponse {
    totalPages: number
    currentPage: number
    files: FileResponse[]
}

export interface FileResponse {
    fileId: number
    link: string
    filename: string
    mimeType: string
    userId?: number
    uploadDate: string
    size: number
    hasThumbnail: boolean
}

File: src/main/kotlin/data/FileResponse.kt
package com.perpheads.files.data

import java.time.Instant

class ListFilesResponse(
    val totalPages: Int,
    val currentPage: Int,
    val files: List<FileResponse>,
)

data class FileResponse(
    val fileId: Int,
    val link: String,
    val filename: String,
    val mimeType: String,
    val uploadDate: Instant,
    val size: Int,
    val thumbnail: String?,
    val hasThumbnail: Boolean
)